source/slang/slang-ir-util.cpp

```diff
@@ -964,6 +967,34 @@ bool isOne(IRInst* inst)
     }
 }
 
+void initializeScratchData(IRInst* inst)
+{
+    List<IRInst*> workList;
```

Comment:
I'm assuming the use of the work list here is to avoid recursive traversal?

Reply:
Right, it is used to implement stackless DFS on the IR module.